src/components/DataGridWithInfiniteScroll/DataGridWithInfiniteScroll.tsx

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) 2022 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import {useTable, Column} from "react-table";
+import {Spinner, SpinnerSize} from "@dellstorage/clarity-react/spinner/Spinner";
+
+/**
+ * General component description :
+ * DataGridWithInfiniteScroll :
+ * Datagrids are for organizing large volumes of data that users can perform actions on.
+ */
+
+/**
+ * @param {rows} rows data
+ * @param {columns} column details
+ * @param {className} CSS class names
+ * @param {style} to add custom CSS styles
+ * @param {isLoading} if true then show loading icon before table is rendered
+ */
+type DataGridProps = {
+    rows: {[key: string]: any}[];
+    columns: Column<{[key: string]: any}>[];
+    className?: string;
+    style?: any;
+    isLoading?: boolean;
+};
+
+function DataGridWithInfiniteScroll(props: DataGridProps) {
+    const columns = props?.columns;
+    const data = props?.rows;
+    const {style, isLoading} = props;
+
+    const {getTableProps, getTableBodyProps, headerGroups, rows, prepareRow} = useTable({
+        columns,
+        data,
+    });
+
+    const renderTable = () => {
+        return (
+            <table {...getTableProps()} style={style}>
+                <thead>
+                    {headerGroups.map((headerGroup) => (
+                        <tr {...headerGroup.getHeaderGroupProps()} className={"csg-header"}>
+                            {headerGroup.headers.map((column) => (
+                                <th {...column.getHeaderProps()}>
+                                    {column.render("Header")}
+                                    <div className={"datagrid-column-separator"} />
+                                </th>
+                            ))}
+                        </tr>
+                    ))}
+                </thead>
```

Comment:
renderTable() will grow in future better create small functions for renderHeader() and renderRow() as well

Reply:
Done